Add the possibility to create snupkg

NielsDM opened the pull request:
At the moment it's only possible to create legacy symbol packages. With this pull request I want to add the possibility to create the more current snupkg packages.

NielsDM committed:
Add the possibility to create snupkg

NielsDM commented:
Hey @mckn,

I'm back with a small update. We should need the possibility to create snupkg packages instead of the legacy symbol.nupkg packages. I added the parameter and tests to pack and push them.

With kind regards,

Niels

NielsDM committed:
Make it possible to overwrite noninteractive

mckn commented:
> Hey @mckn,
> 
> I'm back with a small update. We should need the possibility to create snupkg packages instead of the legacy symbol.nupkg packages. I added the parameter and tests to pack and push them.
> 
> With kind regards,
> 
> Niels

Nice! It looks really good.

mckn commented on lib/argsFor.js:
```diff
@@ -40,7 +42,6 @@ exports.pack = function (nuspecFile, options) {
   });
 
   args.push("-nopackageanalysis");
-  args.push("-noninteractive");
```
What is the reason to remove this? If we want it to be configurable it should probably default to run as `noninteractive` to keep backward compatability.

This is also causing the tests to fail.

NielsDM replied:
Totally correct that it should be backward compatible. I will update this.

We would need to make this a setting because of an authentication popup could popup when pushing to a secure nuget server. A great example is the Azure Devops Artifacts.

mckn commented on lib/argsFor.js:
```diff
@@ -96,7 +104,5 @@ exports.restore = function (nupkg, options) {
     }
   });
 
-  args.push("-noninteractive");
```
Same question here as above.

NielsDM commented:
Thank you! 
Will you be able to merge this PR in the near future? 


Van: "Marcus Andersson" ***@***.***> 
Aan: "mckn/gulp-nuget" ***@***.***> 
Cc: "niels dm" ***@***.***>, "Author" ***@***.***> 
Verzonden: Woensdag 30 maart 2022 11:13:20 
Onderwerp: Re: [mckn/gulp-nuget] Add the possibility to create snupkg (PR #30) 







Hey [ https://github.com/mckn | @mckn ] , 

I'm back with a small update. We should need the possibility to create snupkg packages instead of the legacy symbol.nupkg packages. I added the parameter and tests to pack and push them. 

With kind regards, 

Niels 



Nice! It looks really good. 

— 
Reply to this email directly, [ https://github.com/mckn/gulp-nuget/pull/30#issuecomment-1082824314 | view it on GitHub ] , or [ https://github.com/notifications/unsubscribe-auth/AAXFJM2UAHKTHVZHQOGWL4TVCQLLBANCNFSM5RG6F53A | unsubscribe ] . 
You are receiving this because you authored the thread. Message ID: ***@***.***>

mckn reviewed (commented):
I think everything looks really good! Thanks for opening this PR!

Only things missing:

- Update the README with the new symbols format argument.
- Make sure that we keep backwards compatibility with the noninteractive argument.

mckn commented:
I will be able to merge this as soon as the minor things I mentioned are fixed.

NielsDM committed:
Possible to overwrite the default noninteractive

NielsDM committed:
Remove unnecessary variable

NielsDM commented:
I done the necessary fixes I think:

- The noninteractive can be overriden but the default is still noninteractive
- Extra test for overriding noninteractive
- Added the possibility to override noninteractive to the readme
- Added SymbolPackageFormat to the readme

mckn commented:
Awesome work on this!

NielsDM commented:
You created this great package. Big thanks for that!
